addons/website/static/src/snippets/s_dynamic_snippet/options.js

```diff
@@ -224,10 +227,8 @@ const dynamicSnippetOptions = options.Class.extend({
             if (!this.dynamicFilterTemplates[selectedTemplateId]) {
                 this.$target.get(0).dataset['templateKey'] = dynamicFilterTemplates[0].key;
                 this.isOptionDefault['templateKey'] = true;
-                setTimeout(() => {
-                    this._templateUpdated(dynamicFilterTemplates[0].key, selectedTemplateId);
-                    this._refreshPublicWidgets();
```

Comment:
Should not this be awaited to avoid more race conditions? Would that slow down something?